app/server.js

```diff
@@ -0,0 +1,204 @@
+  res.redirect('/account');
+});
+
+app.get('/account', ensureAuthenticated, co.wrap(function* (req, res) {
+  try { 
+    const user = req.user._json;
+    printMessage('user', user);
+
+    let queryString = `UPDATE user_details set login='${user.login}', github_user_id=${user.id}, html_url='${user.html_url}', repos_url='${user.repos_url}', token='${user.token}' where EXISTS (SELECT * FROM user_details WHERE user_details.github_user_id = ${user.id});
+        INSERT INTO user_details (login, github_user_id, html_url, repos_url, token)  
+        SELECT '${user.login}', ${user.id}, '${user.html_url}', '${user.repos_url}', '${user.token}'
+        WHERE NOT EXISTS (SELECT 1 FROM user_details WHERE user_details.github_user_id = ${user.id});
+
+        SELECT user_details.id, roles.name as roleName FROM user_details, roles
+        WHERE user_details.github_user_id = ${user.id}
+        AND user_details.role_id = roles.id;`;
+
+    user.userDetails = yield dbService.queryDatabase(queryString);
+    req.session.user = user;
+    printMessage('user after log in', user);
+    printMessage('user.userDetails', user.userDetails.rolename)
+    
+    res.render('account', { user, title: 'Moje konto', roleName: user.userDetails.rolename });
+
+  } catch (e) {
+    printMessage('got error', e);
+  }
+}));
+
+
+app.get('/repositories', ensureAuthenticated, (req, res) => {
+
+  const user = req.session.user;
+
+  const storeRepositories = (body) => {
+    let repositories = JSON.parse(body);
+      // `UPDATE repositories set git_project_id='${item.id}', project_name='${item.name}', full_project_name='${item.full_name}', html_url='${item.html_url}', description='${item.description}', api_url='${item.url}'
+      //     where EXISTS (SELECT * FROM repositories WHERE repositories.git_project_id = ${item.id});
+
+    coForEach(repositories, function* (item, index) {
+      let queryString = `INSERT INTO repositories (user_id, git_project_id, project_name, full_project_name, html_url, description, api_url)
+          SELECT '${user.userDetails.id}', '${item.id}','${item.name}', '${item.full_name}', '${item.html_url}', '${item.description}', '${item.url}'
+          WHERE NOT EXISTS (SELECT 1 FROM repositories WHERE repositories.git_project_id = ${item.id});`
+
+          printMessage(queryString);
+
+      yield dbService.queryDatabase(queryString);
+
+    })
+    res.render('repositories', { repositories: repositories });
+  }
+
+  try {
+    requestClient.get(`https://api.github.com/users/${req.user.username}/repos`)
+      .then(storeRepositories)
+  } catch (e) {
+    printMessage('got error', e);
+  }
+
+});
+
+
+app.get(`/repositories/issues/:name`, ensureAuthenticated, (req, res) => {
+
+  const user = req.session.user;
+
+  const storeIssues = (body) => { 
+    let issues = JSON.parse(body); 
+    printMessage(issues);
+    // issues.ForEach((item) => console.log(item));
```

Comment:
byl tam wczesniej update or insert ale nie dzialalo to razem jak tzreba. Zamienilem to na ON CONFLICT
```
`INSERT INTO issues (url, repository_url, git_issue_id, title, user_id, body)
 SELECT '${item.url}', '${item.repository_url}', ${item.id}, '${item.title}', ${user.id}, '${item.body}'
 ON CONFLICT (git_issue_id) DO UPDATE
 SET url='${item.url}', repository_url='${item.repository_url}', git_issue_id='${item.id}', title='${item.title}', user_id='${user.id}', body='${item.body}'`;
```